internal/pkg/term/selector/selector.go

```diff
@@ -196,6 +203,131 @@ type WorkspaceSelector struct {
 	ws     workspaceRetriever
 }
 
+type localFileSelector struct {
+	prompt        prompter
+	fs            *afero.Afero
+	workingDirAbs string
+}
+
+// NewLocalFileSelector constructs a localFileSelector.
+func NewLocalFileSelector(prompt prompter, fs afero.Fs) (*localFileSelector, error) {
+	workingDirAbs, err := os.Getwd()
+	if err != nil {
+		return nil, fmt.Errorf("get working directory: %w", err)
+	}
+	return &localFileSelector{
+		prompt:        prompt,
+		fs:            &afero.Afero{Fs: fs},
+		workingDirAbs: workingDirAbs,
+	}, nil
+}
+
+// Dockerfile asks the user to select from a list of Dockerfiles in the current
+// directory or one level down. If no dockerfiles are found, it asks for a custom path.
+func (s *localFileSelector) Dockerfile(selPrompt, notFoundPrompt, selHelp, notFoundHelp string, pathValidator prompt.ValidatorFunc) (string, error) {
+	dockerfiles, err := s.listDockerfiles()
+	if err != nil {
+		return "", err
+	}
+	var sel string
+	dockerfiles = append(dockerfiles, []string{dockerfilePromptUseCustom, DockerfilePromptUseImage}...)
+	sel, err = s.prompt.SelectOne(
+		selPrompt,
+		selHelp,
+		dockerfiles,
+		prompt.WithFinalMessage(dockerfileFinalMsg),
+	)
+	if err != nil {
+		return "", fmt.Errorf("select Dockerfile: %w", err)
+	}
+	if sel != dockerfilePromptUseCustom {
+		return sel, nil
+	}
+	sel, err = s.prompt.Get(
+		notFoundPrompt,
+		notFoundHelp,
+		pathValidator,
+		prompt.WithFinalMessage(dockerfileFinalMsg))
+	if err != nil {
+		return "", fmt.Errorf("get custom Dockerfile path: %w", err)
+	}
+	return sel, nil
+}
+
+// ListDockerfiles returns the list of Dockerfiles within the current
+// working directory and a sub-directory level below. If an error occurs while
+// reading directories, or no Dockerfiles found returns the error.
+func (s *localFileSelector) listDockerfiles() ([]string, error) {
```

Comment:
This method is copied from https://github.com/aws/copilot-cli/blob/e7fadd11e2bea021a80c977fa210d3fe032b142a/internal/pkg/workspace/workspace.go#L608

I changed the received from `Workspace` to `localFileSelector` because listing/reading Dockerfile does not require the existence of a workspace.

Reply:
What do you think of moving this comment to the `localFileSelector` struct? because it wasn't obvious to me that there was a difference between a workspace and local file selector but it makes sense to me after reading this.

Reply:
Added a comment to the struct!
```
// localFileSelector selects from a local file system where a workspace does not necessarily exist.
type localFileSelector struct {
	prompt        prompter
	fs            *afero.Afero
	workingDirAbs string
}
```